### ISSUE TITLE
Unable to access k8s dashboard

### ISSUE THREAD

**Issue by MirzaSikander** [Azure/aks-engine#453]
**Is this a request for help?**:
Yes
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
Issue
---

**What version of aks-engine?**:
 v0.29.1
---

**Kubernetes version**:
v1.10.12

**What happened**:

When trying to access k8s Web UI dashboard using `kubectl proxy`, you get ` Unauthorized (401)
Authentication failed. Please try again.` 

I already created the following cluster role binding
```
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRoleBinding
metadata:
  name: kubernetes-dashboard
  labels:
    k8s-app: kubernetes-dashboard
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-admin
subjects:
- kind: ServiceAccount
  name: kubernetes-dashboard
```
**What you expected to happen**:
Should be able to access the dashboard. It works on  `k8s v1.10.9` that was deployed using `aks-engine v0.27.0`

**How to reproduce it** (as minimally and precisely as possible):
1) Deploy a brand new cluster with all default settings.
2) Create the Cluster role binding specified above.
3) Run `kubectl proxy`
4) Try to access [http://localhost:8001/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:/proxy/#!/overview?namespace=default](http://localhost:8001/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:/proxy/#!/overview?namespace=default)

**Anything else we need to know**:

**Comment by zhiweiv** [comment]
See https://docs.microsoft.com/en-us/azure/aks/kubernetes-dashboard, **For RBAC-enabled clusters**.

**Comment by MirzaSikander** [comment]
As I mentioned in the description, I already had created that role binding mentioned in that article. Any other suggestion?

**Comment by visitpay** [comment]
This is also an issue for me.  We have the AAD setup, worked in v0.28.1, but doesnt in v0.29.1 and also v0.31.1.  Definite tried:
`kubectl create clusterrolebinding kubernetes-dashboard --clusterrole=cluster-admin --serviceaccount=kube-system:kubernetes-dashboard`

Still gives 401.

**Comment by zhiweiv** [comment]
I tried 1.13, now seems you need to use one service account's bearer toekn to sign in, you should grant cluster-admin to that account, not kubernetes-dashboard.
https://stackoverflow.com/questions/46664104/how-to-sign-in-kubernetes-dashboard

**Comment by visitpay** [comment]
@zhiweiv You are correct.  I was able to get the token to work.

For others:
I deployed:
apiVersion: v1
kind: ServiceAccount
metadata:
  name: admin-user
  namespace: kube-system
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: admin-user
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-admin
subjects:
- kind: ServiceAccount
  name: admin-user
  namespace: kube-system

Then get the token:
kubectl -n kube-system describe secret $(kubectl get secrets -n kube-system -o go-template --template '{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}' | grep admin-user-token) | awk '$1=="token:"{print $2}'

Use that token to authenticate.

Should probably just close this issue as it seems like it's fine.
